aptos-move/framework/aptos-framework/sources/ManagedCoin.move

```diff
@@ -0,0 +1,151 @@
+/// ManagedCoin is built to make a simple walkthrough of the Coins module.
+/// It contains scripts you will need to initialize, mint, burn, transfer coins.
+/// By utilizing this current module, a developer can create his own coin and care less about mint and burn capabilities,
+module AptosFramework::ManagedCoin {
+    use Std::ASCII;
+    use Std::Errors;
+    use Std::Signer;
+
+    use AptosFramework::Coin::{Self, BurnCapability, MintCapability};
+    use AptosFramework::TypeInfo;
+
+    // Errors
+
+    /// When no capabilities (burn/mint) found on an account.
+    const ENO_CAPABILITIES: u64 = 0;
+
+    // Data structures
+
+    /// Capabilities resource storing mint and burn capabilities.
+    /// The resource is stored on the account that initialized coin `CoinType`.
+    struct Capabilities<phantom CoinType> has key {
+        mint_cap: MintCapability<CoinType>,
+        burn_cap: BurnCapability<CoinType>,
+    }
+
+    // Public functions
+
+    /// Withdraw an `amount` of coin `CoinType` from `account` and burn it.
+    public(script) fun burn<CoinType>(
+        account: &signer,
+        amount: u64,
+    ) acquires Capabilities {
+        let type_info = TypeInfo::type_of<CoinType>();
+        let coin_address = TypeInfo::account_address(&type_info);
+
+        assert!(
+            exists<Capabilities<CoinType>>(coin_address),
+            Errors::not_published(ENO_CAPABILITIES),
+        );
+
+        let capabilities = borrow_global<Capabilities<CoinType>>(coin_address);
```

Comment:
Hey,

Not anyone can mint, but anyone can burn.
Really it can be solved by coin developer in his own module, but as it ManagedCoin, I made such option, otherwise BurnCapability becoming useless.

Reply:
I see that makes more sense now. I think this ManagedCoin should potentially be in the examples (or standards) package then instead of being a framework module that gets deployed at 0x1. This can give us flexibility to introduce more standards/extensions that devs can deploy on their own but all make calls to the 0x1::Coin module which contains bare-bone functionalities. For devs/teams that don't need to customize, they can just use the 0x1::Coin module directly.

Reply:
I would prefer that we keep the original intent for ManagedCoin -- that is only the owner has the ability to Mint and Burn.

Reply:
Got it, will change